assets/app/scripts/directives/popups.js

```diff
@@ -86,15 +86,21 @@ angular.module('openshiftConsole')
       restrict: 'E',
       scope: {
         route: '=',
-        service: '='
+        service: '=',
+        warnings: '=' // instead of route and service, can provide an existing set of warnings
       },
       link: function($scope) {
         var updateWarnings = function() {
           var warnings = RoutesService.getRouteWarnings($scope.route, $scope.service);
           $scope.content = warnings.join('<br>');
         };
-        $scope.$watch('route', updateWarnings, true);
-        $scope.$watch('service', updateWarnings, true);
+        if ($scope.route) {
+          $scope.$watch('route', updateWarnings, true);
+          $scope.$watch('service', updateWarnings, true);
+        }
+        else if ($scope.warnings) {
+          $scope.content = $scope.warnings.join('<br>');
```

Comment:
Is this the bug? I suspect you need a `$watch` here. Otherwise, the routes can load before the services, causing the overview warning. Would be nice if we can avoid the warning icon from flickering when that happens.